src/Spout/Writer/XLSX/Manager/WorksheetManager.php

```diff
@@ -253,7 +253,9 @@ private function getCellXML($rowIndexOneBased, $columnIndexZeroBased, Cell $cell
                 $cellXML = '';
             }
         } else {
-            throw new InvalidArgumentException('Trying to add a value with an unsupported type: ' . \gettype($cell->getValue()));
+            $value = $cell->getValueEvenIfError();
+
+            throw new InvalidArgumentException('Trying to add a value with an unsupported type: ' . (\is_object($value) ? \get_class($value) : \gettype($value)));
```

Comment:
I think keeping "object" is clearer here. I'd suggest a combination of both:
```suggestion
$errorMessage = 'Trying to add a value with an unsupported type: ' . \gettype($value);
if (is_object($value)) {
    $errorMessage .= ' (' . get_class($value) . ' given)';
}

throw new InvalidArgumentException($errorMessage);
```

WDYT?

Reply:
I'm fine with that.
The previous implementation was inspired by Symfony codebase - they show only native or classname.